commands/reminder.js

```diff
@@ -0,0 +1,143 @@
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const chrono = require('chrono-node');
+const scheduledScrimSchema = require('../models/scheduled-scrim-schema');
+const db = require('../models/db');
+const config = require('../config.json');
+const { Client, GatewayIntentBits, Partials } = require('discord.js');
+const schedule = require('node-schedule');
+
+const client = new Client({ intents: [GatewayIntentBits.Guilds], partials: [Partials.Channel] });
+
+// Function to send a reminder message in the channel
+async function sendChannelReminder(scheduledScrim) {
+    const channel = await client.channels.fetch(scheduledScrim.channel);
+    const role = scheduledScrim.role;
+    // Send the reminder message in the channel with role ping
+    await channel.send(`<@&${role}>: ${scheduledScrim.content}`);
+}
+
+// Function to schedule reminders from the database at startup
+async function startupReminders() {
+    try {
+        await db; // Wait for the database connection
+
+        // Find all scheduled reminders with a date in the future
+        const futureReminders = await scheduledScrimSchema.find({ date: { $gt: new Date() } });
+
+        console.log('Future Reminders:', futureReminders);
+
+        // Schedule each future reminder
+        futureReminders.forEach((reminder) => {
+            scheduleReminder(reminder, reminder.recurrence);
+        });
+        console.log('Scheduled reminders at startup.');
+    } catch (error) { console.error('Error scheduling reminders at startup:', error); }
+}
+
+client.once('ready', () => {
+    console.log(`Logged in as ${client.user.tag}`);
+    startupReminders();
+});
+
+client.login(config.token);
```

Comment:
This whole first part should be formatted to fit into the events folder more likely. It is not a neccesary part of this command, and can be run using the original client I believe. I might be missing something though.